### PR TITLE
more obvious access control

### PR DESCRIPTION
**Pull request by cainlevy**
This aims to improve the obviousness of access control strategies. I expect more strategies to come, so the pattern here is intentionally flexible.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,35 +1,17 @@
...
-  private def require_trusted_referrer
-    return if Rails.application.config.application_domains.include?(requesting_audience)
-    render status: :forbidden, json: JSONEnvelope.errors('referer' => 'is not a trusted host')
-  end
+  include AccessControl
 
-  private def require_api_credentials
-    auth_strategy = ActionController::HttpAuthentication::Basic
+  # Unauthorized access means credentials are required but absent.
+  class AccessUnauthorized < StandardError; end
```
Comment (cainlevy) [inline app/controllers/application_controller.rb]: Hey, welcome!

I picked these names to mimic the 401 and 403 HTTP codes, but you're quite right that it's still potentially confusing. Will have to give it some thought and maybe consider avoiding those words altogether.

Comment (cainlevy) [inline app/controllers/application_controller.rb]: Yeah, I'm gonna go with `AccessUnauthenticated`. Good spot. 🔍

Comment (cainlevy) [inline app/controllers/application_controller.rb]: https://github.com/keratin/authn/commit/4b2d0022f2f42a4758213736e5caeb0d521f24d2

**Comment by tiegz** [comment]
w00t!

**Pull request merged**